voc/python/ast.py

```diff
@@ -1712,13 +1783,25 @@ def visit_GeneratorExp(self, node):
 
     @node_visitor
     def visit_Yield(self, node):
-        self.visit(node.value)
+        if node.value:
+            self.visit(node.value)
+            self.context.add_opcodes(
+                # Convert to a new value for return purposes
+                JavaOpcodes.INVOKEINTERFACE('org/python/Object', 'byValue', args=[], returns='Lorg/python/Object;')
+            )
+        elif hasattr(node, "lineno"):  # a yield expression
+            # push NoneType object to stack
```

Comment:
It's not clear to me why a None is needed on the stack here - is something consuming it later?

(To that end, comments like "push None onto stack" aren't that helpful - it's fairly easy to see that the next line is "add none to stack"; the comment needs to say *why* that is the required course of action)

Reply:
This is to handle the statement `yield` which is equivalent to `yield None` . I have added a comment to it in latest PR